sktime/forecasting/compose/_dmlf.py

```diff
@@ -0,0 +1,413 @@
+
+        Parameters
+        ----------
+        X : pd.DataFrame
+            Exogenous variables
+
+        Returns
+        -------
+        X_ex : pd.DataFrame
+            Exposure variables
+        X_conf : pd.DataFrame
+            Confounder variables
+        """
+        if X is None:
+            raise ValueError("X cannot be None for DML forecasting")
+
+        # Check that all exposure variables exist in X
+        missing_vars = set(self.exposure_vars) - set(X.columns)
+        if missing_vars:
+            raise ValueError(
+                f"Exposure variables {missing_vars} not found in X columns: "
+                f"{list(X.columns)}"
+            )
+
+        X_ex = X[self.exposure_vars].copy()
+        X_conf = X.drop(columns=self.exposure_vars).copy()
+
+        # Handle case where no confounders remain
+        if X_conf.shape[1] == 0:
+            X_conf = None
+
+        return X_ex, X_conf
+
+    def _fit(self, y, X=None, fh=None):
+        """Fit the DML forecaster.
+
+        Implements the DML fitting procedure:
+        1. Split X into exposure (X_ex) and confounder (X_conf) variables
+        2. Fit forecaster_y on y and X_conf to get outcome residuals
+        3. Fit forecaster_ex on X_ex and X_conf to get exposure residuals
+        4. Fit forecaster_res on the residual relationship
+        5. Fit final versions for prediction
+        """
+        # Split exogenous data
+        X_ex, X_conf = self._split_exogenous_data(X)
+
+        # Step 1: Fit outcome model for residuals (clone A)
+        self.forecaster_y_insample_ = clone(self.forecaster_y)
+        self.forecaster_y_insample_.fit(y, X=X_conf, fh=fh)
+
+        # Get in-sample outcome residuals
+        if isinstance(y.index, pd.MultiIndex):
+            time_idx = y.index.get_level_values(-1).unique()
+        else:
+            time_idx = y.index
+
+        insample_fh = ForecastingHorizon(time_idx, is_relative=False)
+        y_pred_insample = self.forecaster_y_insample_.predict(fh=insample_fh, X=X_conf)
+        self.y_residuals_ = y - y_pred_insample
+
+        # Step 2: Fit exposure model for residuals
+        self.forecaster_ex_insample_ = clone(self.forecaster_ex)
+        self.forecaster_ex_insample_.fit(X_ex, X=X_conf, fh=fh)
+
+        # Get in-sample exposure residuals
+        X_ex_pred_insample = self.forecaster_ex_insample_.predict(
+            fh=insample_fh, X=X_conf
+        )
+        self.X_ex_residuals_ = X_ex - X_ex_pred_insample
+
+        # Step 3: Fit residual relationship
+        self.forecaster_res_ = clone(self.forecaster_res)
+        self.forecaster_res_.fit(self.y_residuals_, X=self.X_ex_residuals_, fh=fh)
+
+        # Step 4: Fit final versions for prediction
+        self.forecaster_y_final_ = clone(self.forecaster_y)
+        self.forecaster_y_final_.fit(y, X=X_conf, fh=fh)
+
+        self.forecaster_ex_final_ = clone(self.forecaster_ex)
+        self.forecaster_ex_final_.fit(X_ex, X=X_conf, fh=fh)
```

Comment:
How are these different from the models fitted earlier? Do we need to do this fitting twice?